cargo-nextest/src/dispatch.rs

```diff
@@ -1797,6 +1702,122 @@ impl App {
         Ok(())
     }
 
+    fn exec_archive(
+        &self,
+        output_file: &Utf8Path,
+        format: ArchiveFormatOpt,
+        zstd_level: i32,
+        output_writer: &mut OutputWriter,
+    ) -> Result<()> {
+        // Do format detection first so we fail immediately.
+        let format = format.to_archive_format(output_file)?;
+        let binary_list = self.base.build_binary_list()?;
+        let path_mapper = PathMapper::noop();
+        let build_platforms = &binary_list.rust_build_meta.build_platforms;
+        let pcx = ParseContext::new(self.base.graph());
+        let (_, config) = self.base.load_config(&pcx)?;
+        let profile = self
+            .base
+            .load_profile(&config)?
+            .apply_build_platforms(build_platforms);
+        let redactor = if should_redact() {
+            Redactor::build_active(&binary_list.rust_build_meta)
+                .with_path(output_file.to_path_buf(), "<archive-file>".to_owned())
+                .build()
+        } else {
+            Redactor::noop()
+        };
+        let mut reporter = ArchiveReporter::new(self.base.output.verbose, redactor.clone());
+
+        if self
+            .base
+            .output
+            .color
+            .should_colorize(supports_color::Stream::Stderr)
+        {
+            reporter.colorize();
+        }
+
+        let rust_build_meta = binary_list.rust_build_meta.map_paths(&path_mapper);
+        let test_artifacts = RustTestArtifact::from_binary_list(
+            self.base.graph(),
+            binary_list.clone(),
+            &rust_build_meta,
+            &path_mapper,
+            self.build_filter.platform_filter.into(),
+        )?;
+
+        let filter_bound = {
+            if self.build_filter.ignore_default_filter {
+                FilterBound::All
+            } else {
+                FilterBound::DefaultSet
+            }
+        };
+
+        let filter_exprs = self.build_filtering_expressions(&pcx)?;
+        let test_filter_builder = self.build_filter.make_test_filter_builder(filter_exprs)?;
+        let ecx = profile.filterset_ecx();
+
+        // Apply filterset to `RustTestArtifact` list.
+        let test_artifacts: BTreeSet<_> = test_artifacts
+            .iter()
+            .filter(|test_artifact| {
+                matches!(
+                    test_filter_builder.filter_binary_match(test_artifact, &ecx, filter_bound),
+                    FilterBinaryMatch::Definite | FilterBinaryMatch::Possible
```

Comment:
Rather than this I'd just reject `test` predicates entirely, returning a `FiltersetParseErrors` that contains all the errors (so you get a nice display). You could do it either at parsing time or as a pass afterwards (have a `check_no_test_predicates` method or similar).

Reply:
Makes sense! I've modified the parser to introduce an error when the filterset kind is `FiltersetKind:TestArchive`.

I've added a test case that verifies an error is displayed if a `test` predicate is used.

Reply:
Just this comment to address, otherwise this is good to go!

Reply:
Sorry missed this previously, just pushed a commit that I believe addresses this.